pdns/dnsdistdist/dnsdist-lua.cc

```diff
@@ -1636,16 +1634,16 @@ static void setupLuaConfig(LuaContext& luaCtx, bool client, bool configCheck)
       const auto pools = dnsdist::configuration::getCurrentRuntimeConfiguration().d_pools;
       for (const auto& entry : pools) {
         const string& name = entry.first;
-        const std::shared_ptr<ServerPool> pool = entry.second;
-        string cache = pool->packetCache != nullptr ? pool->packetCache->toString() : "";
+        const auto& pool = entry.second;
+        string cache = pool.packetCache != nullptr ? pool.packetCache->toString() : "";
         string policy = defaultPolicyName;
-        if (pool->policy != nullptr) {
-          policy = pool->policy->getName();
+        if (pool.policy != nullptr) {
+          policy = pool.policy->getName();
         }
         string servers;
 
-        const auto poolServers = pool->getServers();
-        for (const auto& server : *poolServers) {
+        const auto& poolServers = pool.getServers();
+        for (const auto& server : poolServers) {
```

Comment:
is there a reason not use

```cpp
for (const auto& server : poolServers.getServers()) {..}
```

Or is this to explicitly make the returned servers `const`?

Reply:
No, I'll change it!